src/SpectatorView.Unity/Assets/SpectatorView/Scripts/Compositor/CompositionManager.cs

```diff
@@ -612,10 +612,12 @@ public bool IsRecording()
             return UnityCompositorInterface.IsRecording();
         }
 
-        public void StartRecording()
+        public string StartRecording()
         {
             TextureManager.InitializeVideoRecordingTextures();
-            UnityCompositorInterface.StartRecording((int)VideoRecordingLayout);
+            string filePath = new string(' ', 1024);
+            UnityCompositorInterface.StartRecording((int)VideoRecordingLayout, filePath, filePath.Length);

```

Comment:
If you change the marshalling to use StringBuilder, you'd just create a new StringBuilder(1024) and then pass filePath.Capacity for the length parameter